sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/text/DefaultSource.scala

```diff
@@ -125,6 +126,33 @@ class DefaultSource extends FileFormat with DataSourceRegister {
           }
         }
   }
+
+  override def buildReader(
+      sqlContext: SQLContext,
+      partitionSchema: StructType,
+      dataSchema: StructType,
+      filters: Seq[Filter],
+      options: Map[String, String]): PartitionedFile => Iterator[InternalRow] = {
+    verifySchema(dataSchema)
+
+    val conf = new Configuration(sqlContext.sparkContext.hadoopConfiguration)
+    val broadcastedConf =
+      sqlContext.sparkContext.broadcast(new SerializableConfiguration(conf))
+
+    val unsafeRow = new UnsafeRow(1)
+    val bufferHolder = new BufferHolder(unsafeRow)
+    val unsafeRowWriter = new UnsafeRowWriter(bufferHolder, 1)
```

Comment:
These are not serializable and need to be moved into the the closure.  Also, is this the same as the following?
```scala
val encoder = ExpressionEncoder[String]()
new HadoopFileLinesReader(file, broadcastedConf.value.value).map(encoder.toRow)
```